src/io/export_3mf_v2.cc

```diff
@@ -88,10 +121,39 @@ static bool append_polyset(std::shared_ptr<const PolySet> ps, Lib3MF::PWrapper&
       return true;
     };
 
-    auto triangleFunc = [&](const IndexedFace& indices) -> bool {
+    auto triangleFunc = [&](const IndexedFace& indices, int color_index) -> bool {
       try {
-        Lib3MF::sTriangle t{(Lib3MF_uint32)indices[0], (Lib3MF_uint32)indices[1], (Lib3MF_uint32)indices[2]};
-        mesh->AddTriangle(t);
+        const auto triangle = mesh->AddTriangle({
+          static_cast<Lib3MF_uint32>(indices[0]),
+          static_cast<Lib3MF_uint32>(indices[1]),
+          static_cast<Lib3MF_uint32>(indices[2])
+        });
+
+        if (!ps->colors.empty()) {
+          const auto col = ps->colors[color_index];
+          const auto col_it = ctx.colors.find(col);
+          Lib3MF_uint32 col_idx;
+          if (col_it == ctx.colors.end()) {
+            const Lib3MF::sColor materialcolor{
+              static_cast<Lib3MF_uint8>(256 * col.r()),
+              static_cast<Lib3MF_uint8>(256 * col.g()),
+              static_cast<Lib3MF_uint8>(256 * col.b()),
+              static_cast<Lib3MF_uint8>(256 * col.a())
+            };
```

Comment:
256 * 1.0f cast to a byte becomes 0 on my CPU. How did this ever work?

Reply:
Ha! I was looking at the comparable code in V1 with suspicion.

Reply:
Not sure why that would work in other cases. Synced the implementation for V1 and V2 via dedicated functions.